File: Crio/crio_programming_interview_problems-master/RemoveDups/RemoveDups.java
/*
public class ListNode {
   public int val; 
   public ListNode next;
   public ListNode(int x) { val = x; next = null; }
}
*/
import java.util.*;

import crio.ds.List.ListNode;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

class RemoveDups{
	public static void main(String [] args)
	{
		Scanner sc= new Scanner(System.in);
		int n=sc.nextInt();
		ArrayList<Integer> list = new ArrayList<Integer>();
		for(int i = 0 ; i < n ; ++i) {
            list.add(sc.nextInt());
        }
        sc.close();
        ListNode head = ListNode.createList(list);
        Solution.removeDuplicates(head);
        List<Integer> res = ListNode.extractList(head);
        for(int element: res) {
            System.out.print(element + " ");
        }
        System.out.println();
	}
}

File: Crio/crio_programming_interview_problems-master/RemoveDups/RemoveDups.cpp
#include <bits/stdc++.h>
#include "Solution.cpp"

using namespace std;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main() {
    int n;
    cin >> n;
    vector<int> list(n);
    for(int i = 0 ; i < n ; ++i) {
        cin >> list[i];
    }
    ListNode* head = ListNode().createList(list);

    ListNode* result = removeDups(head);
    vector<int> resultList = ListNode().extractList(result);
    for(int i = 0 ; i < resultList.size() ; ++i) {
        cout << resultList[i] << " ";
    }
    cout << "\n";
    return 0;
}
